contracts/vault/src/query.rs

```diff
@@ -0,0 +1,48 @@
+use perpswap::contracts::vault::QueryMsg;
+
+use crate::{common::get_total_assets, prelude::*, state};
+#[allow(dead_code)]
+/// Handles all queries to the vault
+///
+/// # Parameters
+/// - `deps`: Dependencies for storage access
+/// - `env`: Contract environment
+/// - `msg`: Query message
+///
+/// # Returns
+/// - `StdResult<Binary>`: Serialized query response
+#[entry_point]
+pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::GetVaultBalance {} => {
```

Comment:
This looks surprising. We don't need a smart contract query to look up the balance of an asset for a contract. I would have expected this to include the value of the allocations as well.